lib/smart_proxy_reports/ansible_processor.rb

```diff
@@ -182,4 +209,14 @@ def process_level(result)
       result["level"] = "info"
     end
   end
+
+  def search_for_facts(result)
+    if result.respond_to?(:key?) && result.key?(:ansible_facts)
+      result[:ansible_facts]
+    elsif result.respond_to?(:each)
+      r = nil
+      result.find { |*a| r = search_for_facts(a.last) }
+      r
+    end
+  end
```

Comment:
I don't see this method to be used anywhere, is this a leftover? Also recursive calling smells, why do you need to do it this way? Aren't facts always in the same hierarchy level? Recursive usually has high O(n) and this could cause performance problems.